# This file serves as an example for using the driver with a tf_prefix 'my_ur_'. use this in
# conjunction with
#
# ros2 launch ur_robot_driver ur_control.launch.py ur_type:=ur5e \
# robot_ip:=192.168.56.101 tf_prefix:=my_ur_ launch_rviz:=false \
# controllers_file:=ur_controllers_prefixed.yaml
controller_manager:
  ros__parameters:
    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    io_and_status_controller:
      type: ur_controllers/GPIOController

    speed_scaling_state_broadcaster:
      type: ur_controllers/SpeedScalingStateBroadcaster

    force_torque_sensor_broadcaster:
      type: force_torque_sensor_broadcaster/ForceTorqueSensorBroadcaster

    joint_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    scaled_joint_trajectory_controller:
      type: ur_controllers/ScaledJointTrajectoryController

    forward_velocity_controller:
      type: velocity_controllers/JointGroupVelocityController

    forward_position_controller:
      type: position_controllers/JointGroupPositionController


speed_scaling_state_broadcaster:
  ros__parameters:
    state_publish_rate: 100.0
    tf_prefix: "my_ur_"

io_and_status_controller:
  ros__parameters:
    tf_prefix: "my_ur_"

force_torque_sensor_broadcaster:
  ros__parameters:
    sensor_name: my_ur_tcp_fts_sensor
    state_interface_names:
      - force.x
      - force.y
      - force.z
      - torque.x
      - torque.y
      - torque.z
    frame_id: my_ur_tool0
    topic_name: ft_data


joint_trajectory_controller:
  ros__parameters:
    joints:
      - my_ur_shoulder_pan_joint
      - my_ur_shoulder_lift_joint
      - my_ur_elbow_joint
      - my_ur_wrist_1_joint
      - my_ur_wrist_2_joint
      - my_ur_wrist_3_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    state_publish_rate: 100.0
    action_monitor_rate: 20.0
    allow_partial_joints_goal: false
    constraints:
      stopped_velocity_tolerance: 0.2
      goal_time: 0.0
      my_ur_shoulder_pan_joint: { trajectory: 0.2, goal: 0.1 }
      my_ur_shoulder_lift_joint: { trajectory: 0.2, goal: 0.1 }
      my_ur_elbow_joint: { trajectory: 0.2, goal: 0.1 }
      my_ur_wrist_1_joint: { trajectory: 0.2, goal: 0.1 }
      my_ur_wrist_2_joint: { trajectory: 0.2, goal: 0.1 }
      my_ur_wrist_3_joint: { trajectory: 0.2, goal: 0.1 }


scaled_joint_trajectory_controller:
  ros__parameters:
    joints:
      - my_ur_shoulder_pan_joint
      - my_ur_shoulder_lift_joint
      - my_ur_elbow_joint
      - my_ur_wrist_1_joint
      - my_ur_wrist_2_joint
      - my_ur_wrist_3_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    state_publish_rate: 100.0
    action_monitor_rate: 20.0
    allow_partial_joints_goal: false
    constraints:
      stopped_velocity_tolerance: 0.2
      goal_time: 0.0
      my_ur_shoulder_pan_joint: { trajectory: 0.2, goal: 0.1 }
      my_ur_shoulder_lift_joint: { trajectory: 0.2, goal: 0.1 }
      my_ur_elbow_joint: { trajectory: 0.2, goal: 0.1 }
      my_ur_wrist_1_joint: { trajectory: 0.2, goal: 0.1 }
      my_ur_wrist_2_joint: { trajectory: 0.2, goal: 0.1 }
      my_ur_wrist_3_joint: { trajectory: 0.2, goal: 0.1 }
    speed_scaling_interface_name: my_ur_speed_scaling/speed_scaling_factor

forward_velocity_controller:
  ros__parameters:
    joints:
      - my_ur_shoulder_pan_joint
      - my_ur_shoulder_lift_joint
      - my_ur_elbow_joint
      - my_ur_wrist_1_joint
      - my_ur_wrist_2_joint
      - my_ur_wrist_3_joint
    interface_name: velocity

forward_position_controller:
  ros__parameters:
    joints:
      - my_ur_shoulder_pan_joint
      - my_ur_shoulder_lift_joint
      - my_ur_elbow_joint
      - my_ur_wrist_1_joint
      - my_ur_wrist_2_joint
      - my_ur_wrist_3_joint
